tests/test_whitelist.py

```diff
@@ -1,37 +1,605 @@
 from tests.module_factory import ModuleFactory
 import pytest
+import json
+from unittest.mock import MagicMock, patch
+from slips_files.core.evidence_structure.evidence import (
+    Direction,
+    IoCType
+)
+from conftest import mock_db
+import os
 
 
-def test_read_whitelist(mock_db):
+def test_read_whitelist(
+        mock_db
+):
     """
     make sure the content of whitelists is read and stored properly
     uses tests/test_whitelist.conf for testing
     """
     whitelist = ModuleFactory().create_whitelist_obj(mock_db)
     mock_db.get_whitelist.return_value = {}
-    (
-        whitelisted_IPs,
-        whitelisted_domains,
-        whitelisted_orgs,
-        whitelisted_mac,
-    ) = whitelist.read_whitelist()
-    assert "91.121.83.118" in whitelisted_IPs
-    assert "apple.com" in whitelisted_domains
-    assert "microsoft" in whitelisted_orgs
+    whitelisted_IPs, whitelisted_domains, whitelisted_orgs, whitelisted_mac = whitelist.read_whitelist()
+    assert '91.121.83.118' in whitelisted_IPs
+    assert 'apple.com' in whitelisted_domains
+    assert 'microsoft' in whitelisted_orgs
 
 
-@pytest.mark.parametrize("org,asn", [("google", "AS6432")])
-def test_load_org_asn(org, asn, mock_db):
+@pytest.mark.parametrize('org,asn', [('google', 'AS6432')])
+def test_load_org_asn(org, asn,
+                      mock_db
+                      ):
     whitelist = ModuleFactory().create_whitelist_obj(mock_db)
     assert whitelist.load_org_asn(org) is not False
     assert asn in whitelist.load_org_asn(org)
 
 
-@pytest.mark.parametrize("org,subnet", [("google", "216.73.80.0/20")])
-def test_load_org_IPs(org, subnet, mock_db):
+def test_load_org_IPs(mock_db):
     whitelist = ModuleFactory().create_whitelist_obj(mock_db)
-    assert whitelist.load_org_IPs(org) is not False
-    # we now store subnets in a dict sorted by the first octet
-    first_octet = subnet.split(".")[0]
-    assert first_octet in whitelist.load_org_IPs(org)
-    assert subnet in whitelist.load_org_IPs(org)[first_octet]
+    org_info_file = os.path.join(whitelist.org_info_path, 'google')
+    with open(org_info_file, 'w') as f:
+        f.write('34.64.0.0/10\n')
+        f.write('216.58.192.0/19\n')
+
+    org_subnets = whitelist.load_org_IPs('google')
+    assert '34' in org_subnets
+    assert '216' in org_subnets
+    assert '34.64.0.0/10' in org_subnets['34']
+    assert '216.58.192.0/19' in org_subnets['216']
+    os.remove(org_info_file)
+
+
+@pytest.mark.parametrize(
+    "mock_ip_info, mock_org_info, ip, org, expected_result", [
+        ({'asn': {'asnorg': 'microsoft'}}, [json.dumps(['microsoft']), json.dumps([])], "91.121.83.118", "microsoft",
+         True),
+        ({'asn': {'asnorg': 'microsoft'}}, [json.dumps(['microsoft']), json.dumps([])], "91.121.83.118", "apple", True),
+        ({'asn': {'asnorg': 'Unknown'}}, json.dumps(['google']), "8.8.8.8", "google", None),
+        ({'asn': {'asnorg': 'AS6432'}}, json.dumps([]), "8.8.8.8", "google", None),
```

Comment:
adding comments to your test cases would make it way easier to test and understand what they do, i had to use a debugger to test this one :'D, something like "#testing the case where we dont have ASN info about an org" would be awesome